lib/ffprobe.ex

```diff
@@ -14,11 +14,17 @@ defmodule FFprobe do
 
   @doc """
   Get the duration in seconds, as a float.
-  If no duration (e.g., a still image), returns `:no_duration`
+  If no duration (e.g., a still image), returns `:no_duration`.
+  If the file does not exist, returns { :error, :no_such_file } 
   """
   @spec duration(binary | format_map) :: float | :no_duration
   def duration(file_path) when is_binary(file_path) do
-    duration(format(file_path))
+    with format = %{} <- format(file_path) do
```

Comment:
Not sure if I'm a fan of using `with` for single assignments and such short functions, instead prefer `case`